integr8ly/installation
# Middleware monitoring config

Rebase of #385 on #384 

@sedroche 

## Verification Steps for 3scale

1. Run this installation branch.
2. Go to the Prometheus UI. (`prometheus-route`/config  in the `middleware-monitoring` namespace) and make sure a new config is present.

![config](https://user-images.githubusercontent.com/22113654/53110689-74c23680-3533-11e9-8c81-00265baf9d3d.png)

3. Ensure there are 2 new targets at `prometheus-route`/targets
![targets](https://user-images.githubusercontent.com/22113654/53110501-f36aa400-3532-11e9-88cd-9ccd02c1ebe6.png)
4. Ensure you can run a query v the new targets. e.g. `nginx_http_connections{_3scale_component_element="staging"}` at `prometheus-route`/graph
![graph](https://user-images.githubusercontent.com/22113654/53110622-417fa780-3533-11e9-98f1-649621804ca0.png)

## Verification Steps for federation

1. Run the installer and ensure it completes (should already be done in 3scale)
2. Ensure Kube State Metrics is not in the `middleware-monitoring` namespace
3. Ensure an `openshift-monitoring-federation` target exists in prometheus and is up
![screen shot 2019-02-21 at 10 27 17](https://user-images.githubusercontent.com/8698527/53162657-64a66780-35c4-11e9-8754-f0a47818d735.png)
4. Ensure Alerts are passing (they may be pending for a while)
<img width="1615" alt="screen shot 2019-02-08 at 21 15 24" src="https://user-images.githubusercontent.com/8698527/53162665-68d28500-35c4-11e9-8daa-54e178911d63.png">
5. Run the uninstaller and ensure it completes
